### ISSUE TITLE
Error "Current working directory is not a Cordova-based project" when it definitely is!

### ISSUE THREAD

**Issue by atuttle** [phonegap/phonegap-cli#601]
I was able to compile this app for iOS a few weeks ago (just 14-21 days!), but put off compiling for Android until now because the wait time for publishing is basically nothing. And now this! 😭 And of course I need it today.

It's been a few weeks so a few things about my machine could have changed, but I most definitely did not upgrade node. I _might_ have upgraded npm but I don't recall for sure.

```
$ node --version && npm --version && phonegap --version
v6.1.0
3.8.6
(node:42356) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
6.2.5
```

![image](https://cloud.githubusercontent.com/assets/46990/15541130/f23ed67c-2258-11e6-8a71-6fb8b97b3c43.png)

I don't know what triggers the "Current working directory is not a Cordova-based project" error, but I have another similar app, structured the same way, for which `phonegap prepare android` works fine!

![image](https://cloud.githubusercontent.com/assets/46990/15541073/9f587aee-2258-11e6-909b-f419895d92d0.png)

**Comment by atuttle** [comment]
I was able to get a build together by starting a "new project" (`phonegap create`), installing the various plugins and platforms, and copying my content & config... but what a pain.

I tried to debug the issue in the CLI but couldn't find where this message was generated and gave up. :-\

**Comment by mwbrooks** [comment]
Hey @atuttle, let's see if we can fix this issue for you!

The `Error: Current working directory is not a Cordova-based project.` is [called here](https://github.com/apache/cordova-lib/blob/441fd90e8a9e7e647ca3c19a796bbde01734cc50/cordova-lib/src/cordova/util.js#L128) and [this is the function](https://github.com/apache/cordova-lib/blob/441fd90e8a9e7e647ca3c19a796bbde01734cc50/cordova-lib/src/cordova/util.js#L94-L122) that evaluates whether it is a Cordova project and [here is the criteria](https://github.com/apache/cordova-lib/blob/441fd90e8a9e7e647ca3c19a796bbde01734cc50/cordova-lib/src/cordova/util.js#L73-L89) that it uses.

Basically, it looks like it expects:

- `www/`
- platforms/`
- `config.xml` or `www/config.xml`

For your project:

- All of those files and directories exist
- Permissions look good

What stands out to me is that the `www/` is a symbolic link. In the past, I've had success using sym-links, but perhaps that's causing the issue or the linked directory has an odd permission?

**Comment by atuttle** [comment]
The symlink is created using: `phonegap create [...] --link-to elsewhere/`, and as you can see from my screen shots, both the working and non-working directories use symlinks. I can appreciate the fact that this may be an issue, though. Symlinks cause so many problems! :)

**Comment by purplecabbage** [comment]
If this has been answered to your satisfaction, please close it.
Thank you for bringing this to our attention @atuttle

**Comment by atuttle** [comment]
I mean... Maybe? I was able to work around the issue by recreating the project, but as far as I can tell the underlying issue remains -- that symlinks can be unpredictable for this type of thing.

If you want it closed, that's fine with me. I've got a useable, if unpleasant, workaround for the future when this will surely crop up again. But I could also see the case for leaving it open as a reminder to investigate symlink issues.

**Comment by purplecabbage** [comment]
Do you know for sure that you did not destroy the contents of the symlink? because that would of course fail.  
Otherwise, we need steps to reproduce the issue.
Next time it happens, please try a `ls www/` to see if it is still valid.

**Comment by atuttle** [comment]
Dang, that does indeed appear to have been the problem. It never occurred to me that the directory structure could have changed. And in fact it did. Funny how our brains can block out seemingly insignificant details of the recent history.

**Comment by purplecabbage** [comment]
:) 
Thanks for following up!

**Comment by mix3d** [comment]
adding my comment here; the `www` folder is apparently super important. I have a script generating my www folder (due to platform asset differences at build time), so sharing the project didn't have a www folder in Git, and my colleague was getting this error. This thread helped me fix that.

**Comment by acikrahim** [comment]
If problem still persist, try to remove anything inside your /platforms if any. Then, re-add your preferred platform again.

**Comment by PostImpatica** [comment]
I found that I can get this after doing an `ng build` to my cordova directory but cancelling early.  The `ng build` apparently deletes the `www` folder first before deploying.  The folder isn't re-created until it's finished building.
